source/utils/LogManager.h

```diff
@@ -18,45 +18,64 @@
 
 #include <OgreSingleton.h>
 
-#define OD_STRING_S(x) #x
-#define OD_STRING_S_(x) OD_STRING_S(x)
-#define S__LINE__ OD_STRING_S_(__LINE__)
-#define OD_LOG_ERR(a)      LogManager::getSingleton().logMessage("ERROR: " + std::string(__FILE__) + " line " + std::string(S__LINE__) + std::string(" ") + (a), LogMessageLevel::CRITICAL)
-#define OD_LOG_WRN(a)      LogManager::getSingleton().logMessage("WARNING: " + std::string(__FILE__) + " line " + std::string(S__LINE__) + std::string(" ") + (a), LogMessageLevel::WARNING)
-#define OD_LOG_INF(a)      LogManager::getSingleton().logMessage(a, LogMessageLevel::NORMAL)
-#define OD_LOG_DBG(a)      LogManager::getSingleton().logMessage(a, LogMessageLevel::TRIVIAL)
-#define OD_ASSERT_TRUE(a)        if(!(a)) LogManager::getSingleton().logMessage("ERROR: Assert failed file " + std::string(__FILE__) + " line " + std::string(S__LINE__), LogMessageLevel::CRITICAL)
-#define OD_ASSERT_TRUE_MSG(a,b)  if(!(a)) LogManager::getSingleton().logMessage("ERROR: Assert failed file " + std::string(__FILE__) + " line " + std::string(S__LINE__) + std::string(" info : ") + b, LogMessageLevel::CRITICAL)
+#include "utils/Helper.h"
+#include "utils/LogMessageLevel.h"
+#include "utils/LogSink.h"
 
+#ifndef OD_LOG_ALLOW_DEPRECATED
+    #define OD_LOG_ALLOW_DEPRECATED 1
+#endif
 
-enum class LogMessageLevel
-{
-    TRIVIAL = 1,
-    NORMAL,
-    WARNING,
-    CRITICAL
-};
+#if OD_LOG_ALLOW_DEPRECATED
+    // DEPRECATED: Use OD_TRACE_ERR instead.
+    #define OD_LOG_ERR(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+    // DEPRECATED: Use OD_TRACE_WRN instead.
+    #define OD_LOG_WRN(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::WARNING, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+    // DEPRECATED: Use OD_TRACE_INF instead.
+    #define OD_LOG_INF(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+    // DEPRECATED: Use OD_TRACE_DBG instead.
+    #define OD_LOG_DBG(message)                      LogManager::getSingleton().logMessage(LogMessageLevel::TRIVIAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+
+    // DEPRECATED: Use OD_ASSERT instead.
+    #define OD_ASSERT_TRUE(condition)                if (!(condition)) LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "", __FILE__, __LINE__, #condition, 0)
+    // DEPRECATED: Use OD_ASSERT_MSG instead.
+    #define OD_ASSERT_TRUE_MSG(condition, message)   if (!(condition)) LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, "", __FILE__, __LINE__, (std::string("") + message).c_str(), 0)
+#endif
+
+#define OD_TRACE_ERR(_module, _message, ...)         LogManager::getSingleton().logMessage(LogMessageLevel::CRITICAL, #_module, __FILE__, __LINE__, _message, __VA_ARGS__)
```

Comment:
Implemented. ;) The module name is now the filename with the extension, so log messages can be copied between files without prejudice.